sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala

```diff
@@ -1235,7 +1235,7 @@ class HiveDDLSuite
     if (tableExists && !cascade) {
       assertAnalysisError(
         sqlDropDatabase,
-        s"Cannot drop a non-empty database: $dbName.")
+        s"Cannot drop a non-empty namespace: $dbName.")
```

Comment:
My concern is that this may confuse traditional SQL users, as `namespace` is a new concept to them.

I think we need to allow catalog implementations to customize the alias of namespace, such as database, schema. We can add a new API to `CatalogPlugin`
```
default String namespaceAlias() { return "namespace"; };
```

Reply:
Updated. Could you please take a look? Sorry about my mistake, I reversed the test of `HiveDDLSuite`. Do we still need to add a new API to `CatalogPlugin`: `namespaceAlias`? @cloud-fan

Reply:
how do we keep the error message unchanged after switching to v2 DROP NAMESPACE?

Reply:
The above snippet calls `dropDatabase` in `HiveExternalCatalog`, throwing `QueryCompilationErrors.cannotDropNonemptyDatabaseError(db)`. This was included in [#PR 35007](https://github.com/apache/spark/pull/35007). @cloud-fan